clients/drcachesim/tracer/func_trace.h

```diff
@@ -37,15 +37,24 @@
 
 #include "trace_entry.h"
 
-typedef void (*func_trace_append_entry_t)(void *drcontext, trace_marker_type_t marker,
-                                          uintptr_t value);
-typedef void (*func_trace_memtrace_if_redzone_t)(void *drcontext);
+#define MAX_FUNC_TRACE_ENTRY_VEC_CAP 16
+
+typedef struct {
+    trace_marker_type_t marker_type;
+    uintptr_t marker_value;
+} func_trace_entry_t;
+
+typedef struct {
+    int size;
+    func_trace_entry_t entries[MAX_FUNC_TRACE_ENTRY_VEC_CAP];
```

Comment:
Maybe we could use a drvector_t instead of a hardcoded array, so it can dynamically size?

Or just add a comment about potentially doing that.

Reply:
Will add a comment about this.